megamek/src/megamek/client/bot/princess/PathRanker.java

```diff
@@ -172,13 +194,6 @@ abstract RankedPath rankPath(MovePath path, IGame game, int maxRange, double fal
             	
                 Coords finalCoords = path.getFinalCoords();
 
-                // If fleeing, skip any paths that don't get me closer to home.
-                if (fleeing && (distanceToHomeEdge(finalCoords, homeEdge, game) >= startingHomeDistance)) {
```

Comment:
This situation is now handled by generating long-range paths, which are "guaranteed" to move the unit in the direction of the destination, unless it can't get there anyway.